zocalo_dls/wrapper/generic.py

```diff
@@ -17,6 +17,12 @@ def run(self):
         result = procrunner.run(command)
         logger.info("Command successful, took %.1f seconds", result["runtime"])
 
+        if result["exitcode"]:
+            self.failure()
+            return
+        else:
+            self.success()
```

Comment:
The `.success()` and `.failure()` functions are called from the `zocalo.wrap` command based on the truthiness of the `return` value of `run()`.

In simple terms: replacing `return True` with `return not result["exitcode"]` at the end of this function is more likely to do what you want to do.